Yosemite/Yosemite/Stores/ProductShippingClassStore.swift

```diff
@@ -40,22 +40,23 @@ private extension ProductShippingClassStore {
 
     /// Synchronizes the `ProductShippingClass`s associated with a given Site ID (if any!).
     ///
-    func synchronizeProductShippingClassModels(siteID: Int64, pageNumber: Int, pageSize: Int, onCompletion: @escaping (Error?) -> Void) {
+    func synchronizeProductShippingClassModels(siteID: Int64, pageNumber: Int, pageSize: Int, onCompletion: @escaping (Result<Bool, Error>) -> Void) {
         let remote = ProductShippingClassRemote(network: network)
 
-        remote.loadAll(for: siteID, pageNumber: pageNumber, pageSize: pageSize) { [weak self] (models, error) in
-            guard let models = models else {
-                onCompletion(error)
-                return
-            }
-
-            if pageNumber == Default.firstPageNumber {
-                self?.deleteStoredProductShippingClassModels(siteID: siteID)
-            }
-
-            self?.upsertStoredProductShippingClassModelsInBackground(readOnlyProductShippingClassModels: models,
-                                                                     siteID: siteID) {
-                                                                        onCompletion(nil)
+        remote.loadAll(for: siteID, pageNumber: pageNumber, pageSize: pageSize) { [weak self] result in
+            switch result {
+            case .failure(let error):
+                onCompletion(.failure(error))
+            case .success(let models):
+                if pageNumber == Default.firstPageNumber {
+                    self?.deleteStoredProductShippingClassModels(siteID: siteID)
+                }
+
+                self?.upsertStoredProductShippingClassModelsInBackground(readOnlyProductShippingClassModels: models,
+                                                                         siteID: siteID) {
+                                                                            let hasNextPage = models.count == pageSize
+                                                                            onCompletion(.success(hasNextPage))
```

Comment:
This might be a good opportunity to add `guard let self = self` so that we can be sure that both `deleteStoredProductShippingClassModels` and `upsertStoredProductShippingClassModelsInBackground` are executed as **all or nothing**.

Reply:
yup this certainly sounds more robust! updated in https://github.com/woocommerce/woocommerce-ios/pull/2748/commits/37fbfe7bce3667ee25ffda170a17351f38eb2718